pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -1154,3 +1180,441 @@ func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx cont
 
 	return false, nil
 }
+
+func TestPopulateDesiredDefaultParamSet(t *testing.T) {
+	desiredDefaultParamSet := newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange})
+	invalidDefaultParamSet := newL3GNP(networkv1.DefaultPodNetworkName, reconcileMode, nil)
+
+	tests := []struct {
+		name            string
+		defaultParamSet *networkv1.GKENetworkParamSet
+		wantParamSet    *networkv1.GKENetworkParamSet
+	}{
+		{
+			name: "not populate if addon is reconcile mode",
+			defaultParamSet: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+					Annotations: map[string]string{
+						annotationComponentsLayer: componentLayer,
+						annotationComponentsName:  componentName,
+					},
+					Labels: map[string]string{
+						labelsAddonManagerMode: reconcileMode,
+					},
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:       defaultTestNetworkName,
+					VPCSubnet: defaultTestSubnetworkName,
+				},
+			},
+			wantParamSet: invalidDefaultParamSet,
+		},
+		{
+			name: "revert vpc",
+			defaultParamSet: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+					Annotations: map[string]string{
+						annotationComponentsLayer: componentLayer,
+						annotationComponentsName:  componentName,
+					},
+					Labels: map[string]string{
+						labelsAddonManagerMode: ensureExistsMode,
+					},
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:           "random-vpc",
+					VPCSubnet:     defaultTestSubnetworkName,
+					PodIPv4Ranges: &networkv1.SecondaryRanges{RangeNames: []string{defaultPodRange}},
+				},
+			},
+			wantParamSet: desiredDefaultParamSet,
```

Comment:
Well, ideally, we never want to do this. It indicates a bug somewhere else in our system.

Reply:
Sounds fair to me.